hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1145,6 +1145,10 @@ public String getKeyGeneratorClass() {
     return getString(KEYGENERATOR_CLASS_NAME);
   }
 
+  public boolean autoGenerateRecordKeys() {
+    return getBooleanOrDefault(HoodieTableConfig.AUTO_GENERATE_RECORD_KEYS);
+  }
```

Comment:
Can we eliminate this option? We can infter keyless ingestion from 2 other options:

1. the write operation is defined as `INSERT`;
2. or user does not declare any `record.key` fields.

Reply:
yes, I had a jam w/ vinoth yesterday. we are going to rely on user configuring record key. If its configured we will honor it. if not, we will auto generate it. wrt operation type, its little involved. We have to decide the default operation type based on whether use has configured other props or not. 

my high level goal here is:
for someone using plain parquet (writing to parquet table), when they switch to hudi, we want to give them smoother transition. 

so, df.wirte.format("parquet").save(location)
should be easily translatable to 
df.wirte.format("hudi").save(newLocation).

along w/ it, we can infer partitions if 
df.write.partitionBy(colA).format("hudi").save(newLocation). 

So, in these cases(no precombine), we should make insert as the default operation type. In other cases, we will make upsert as default operation type (only when user has not overridden the operation type).